client.go

```diff
@@ -437,6 +445,27 @@ func (c *client) Range(ctx context.Context, r *RangeOp) ([]DomainObject, string,
 	return objectArray, token, nil
 }
 
+func (c *client) FetchRange(ctx context.Context, r *RangeOp, onNext func(value DomainObject) error) error {
```

Comment:
If the inner func wants to make another database call, it will need the context passed to it